utils/src/main/java/com/cloud/utils/nio/NioConnection.java

```diff
@@ -125,7 +125,7 @@ public boolean isStartup() {
     public Boolean call() throws NioConnectionException {
         while (_isRunning) {
             try {
-                _selector.select();
+                _selector.select(1000);
```

Comment:
@swill this change ^^ causes the selector loop to never block indefinitely but at most 1 second. This ensures that reconnections are fast. If anyone experiences the behaviour that tel-netting to port 8250 causes agent to reconnect then, either (1) they are not using latest master or (2) they did not wait for up to one second (i.e. hit an edge case), in which case we can lower this number to 100-500 millisecond at the expense of increasing cpu usage.